### ISSUE TITLE
Builders are too picky about blockstates

### ISSUE THREAD

**Issue by SanderKnauff** [ldtteam/minecolonies#6649]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.4
- MineColonies Version: 667
- Structurize Version: 144
- Related Mods and their Versions: Direwolf20 1.16 1.2.0 with manually updated Minecolony and Structurize

## Expected behavior
Builders should be able to place blocks without them getting confused about the block which just has been placed. 

## Actual behavior
Builders get stuck placing blocks like fences next to other blocks or corner stairs. The builder Wil keep attempting to "fix" said block without succes, causing building progress to effectively halt. Builders will also try to "fix" doors that have been opened during building with the same outcome. 

![109731235-284a7080-7bbb-11eb-8341-c5b3650c98f9](https://user-images.githubusercontent.com/7213734/109737143-d65b1800-7bc5-11eb-9ae8-d634e0acd425.png)
![109731228-2680ad00-7bbb-11eb-9fb1-d9586c4dfea1](https://user-images.githubusercontent.com/7213734/109737155-d9ee9f00-7bc5-11eb-9a33-0dc70f445fb0.png)

## Steps to reproduce the issue

1. Given a town exists with an available builder
2. Try to build either a level 1 MedivalOakAlternative restaurant or a level 1 Caledonia builder hut
3. The builder will get stuck placing fences and/or stairs and will not finish the construction 

### Logs

I do not have access to the logs at the moment, if requested I will reproduce this and provide them. 

### Notes
Original report from the general schematics issue:

Builders are too picky with the blockstates on the MedivalOakAlternative theme. When building a restaurant with Minecolony 667 and Structurize 144, the builder will get stuck when placing fences or stairs for example.

Placing them potentially changes the blockstate (e.g. for fences the side they connect to a block) which seems to confuse the Builder.

As a result, the builder keeps attempting to place a fence in the same place one already is in, without result as it immediately goes into a, for the builder, unexpected blockstate.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
I'll look into this tomorrow again.

**Comment by Talyda** [comment]
I *think* this is related - was testing upgrading a level 2 stone courier hut to level 3, to see if it stopped working at the same place as another person that was having trouble with it (it didn't stop at the same point) but what I did notice, is while the builder stood there whacking at the air with a bed and not actually doing anything, if I opened the door he immediately closed it, over and over again. If I broke the door he immediately stopped everything and went back to his hut and requested a door. I tried replacing it, and leaving it open, to see if that would trick him, but he just closes it and whacks the air some more.
Also tried to remove the stone blocks from either side of the door at the bottom to get the door to break just to see what would happen, but the moment I break one of them, he replaces it.
